lib/python/ensembl/brc4/runnable/download_assembly_data.py

```diff
@@ -21,25 +21,36 @@
 import re
 import ftplib
 import hashlib
+from pathlib import Path
 
+FILE_ENDS = {
+    "assembly_report.txt": "report",
+    "genomic.fna.gz": "fasta_dna",
+    "protein.faa.gz": "fasta_pep",
+    "genomic.gff.gz": "gff3_raw",
+    "genomic.gbff.gz": "gbff",
+}
+    
 
 class download_assembly_data(eHive.BaseRunnable):
-    
+
     @staticmethod
     def param_defaults() -> Dict[str, Any]:
         return {
             # Set this manually to a higher value if you want to allow assembly versions
             # higher than the one provided (it will fail if the version given is not the latest)
             "max_increment": 0,
+            # Set max number of times to retry downloading a file
+            "max_redo": 3,
         }
 
     def run(self):
         accession = self.param_required('accession')
         main_download_dir = self.param_required('download_dir')
-        download_dir = main_download_dir + "/" + accession
+        download_dir = Path(main_download_dir, accession)
 
         # Set and create dedicated dir for download
-        if not os.path.isdir(download_dir):
+        if not download_dir.exists():
```

Comment:
```suggestion
        if not download_dir.is_dir():
```
I think this option meets better the previous implementation.